python/pyproject.toml

```diff
@@ -39,7 +39,10 @@ packages = [
   { include = "Makefile", format = "sdist" },
   { include = "NOTICE", format = ["sdist", "wheel"] }
 ]
-include = [{ path = "dev", format = "sdist" }]
+include = [
+  { path = "dev", format = "sdist" },
+  { path = "pyiceberg/**/*.so", format = "wheel" }
```

Comment:
I pulled in this branch on top of https://github.com/apache/iceberg/pull/8287 and it worked like a charm 🥳

Reply:
yay!